src/Compilers/CSharp/Portable/CodeGen/EmitAddress.cs

```diff
@@ -104,14 +104,8 @@ private LocalDefinition EmitAddress(BoundExpression expression, AddressKind addr
                     break;
 
                 case BoundKind.AssignmentOperator:
-                    var assignment = (BoundAssignmentOperator)expression;
-                    if (assignment.RefKind == RefKind.None)
-                    {
-                        goto default;
-                    }
-
-                    EmitAssignmentExpression(assignment, UseKind.UsedAsAddress);
-                    break;
+                    Debug.Assert(((BoundAssignmentOperator)expression).RefKind == RefKind.None);
```

Comment:
This is to support byref assignments that are not a part of a declaration. Currently they are not supported by the language.

This particular part is for cases like
```
ref var x = ref y = ref z;     // ref-assigning a result of a ref assignment.
```